### ISSUE TITLE
Extract open_new_version  from bulk jobs

### ISSUE THREAD

**Issue by jcoyne** [sul-dlss/argo#3163]
At least a few jobs open a new version if needed.  It would be nice to extract this bit of common features into a method:

```ruby
      state_service = StateService.new(cocina_object.externalIdentifier, version: cocina_object.version)
      open_new_version(cocina_object.externalIdentifier, cocina_object.version, 'message') unless state_service.allows_modification?
      
      
```

**Comment by aaron-collier** [comment]
Moving this back to ready while I'm on vacation for the next 2 weeks. The open PR linked above has where I've left off. The pattern here is to move instantiation of the `StateService` into the `open_new_version` method (https://github.com/sul-dlss/argo/pull/3319/files#diff-17ccf9399eaf1bb82b122d501ef9582ba97c60c32a174e83027885404173088bR72) and return a cocina object instead of a version number to use to update the existing cocina object (the main repeating pattern we want to DRY up here)

**Comment by jmartin-sul** [comment]
i think that was the verdict from @justinlittman in post-standup discussion (i noticed the work in #3319 seemed obsolete when i started rebasing it again).  looks like the refactoring of the `open_version` method from this issue's (unmerged) PR has already been merged to `main` via another PR:
* https://github.com/sul-dlss/argo/pull/3319/files#diff-17ccf9399eaf1bb82b122d501ef9582ba97c60c32a174e83027885404173088bR113
* https://github.com/sul-dlss/argo/pull/3535/files#diff-17ccf9399eaf1bb82b122d501ef9582ba97c60c32a174e83027885404173088bR159

closing this and the associated PR, but happy to reopen and follow through with the rebase if that's incorrect.

**Comment by jmartin-sul** [comment]
closed by #3535
